eZ/Bundle/EzPublishIOBundle/Command/MigrateFilesCommand.php

```diff
@@ -0,0 +1,276 @@
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if ($input->getOption('list-io-configs')) {
+            $this->outputConfiguredHandlers($output);
+
+            return;
+        }
+
+        $output->writeln($this->getProcessedHelp());
+
+        $fromHandlers = $input->getOption('from') ? explode(',', $input->getOption('from')) : null;
+        $toHandlers = $input->getOption('to') ? explode(',', $input->getOption('to')) : null;
+        if (!$this->areHandlerOptionsValid($fromHandlers, $toHandlers, $output)) {
+            return;
+        }
+
+        if (!$fromHandlers) {
+            $fromHandlers = ['default', 'default'];
+        }
+        if (!$toHandlers) {
+            $toHandlers = [
+                array_keys($this->configuredMetadataHandlers)[0],
+                array_keys($this->configuredBinarydataHandlers)[0],
+            ];
+        }
+
+        $output->writeln([
+            "Migrating from '$fromHandlers[0],$fromHandlers[1]' to '$toHandlers[0],$toHandlers[1]'",
+            '',
+        ]);
+        $this->migrationHandler->setIODataHandlersByIdentifiers(
+            $fromHandlers[0],
+            $fromHandlers[1],
+            $toHandlers[0],
+            $toHandlers[1]
+        );
+
+        $totalCount = $this->migrationHandler->countFiles();
+        $output->writeln([
+            'Found total files to update: ' . $totalCount,
+            '',
+        ]);
+
+        if ($totalCount === 0) {
+            $output->writeln('Nothing to process.');
+
+            return;
+        }
+
+        $helper = $this->getHelper('question');
+        $question = new ConfirmationQuestion(
+            '<question>Are you sure you want to proceed?</question> ',
+            false
+        );
+
+        if (!$helper->ask($input, $output, $question)) {
+            $output->writeln('Aborting.');
+
+            return;
+        }
+
+        $bulkCount = $input->getOption('bulk-count');
+        $dryRun = $input->getOption('dry-run');
+
+        $this->migrateFiles(
+            $totalCount,
+            $bulkCount,
+            $dryRun,
+            $output
+        );
+    }
+
+    /**
+     * Output the configured meta/binary data handlers.
+     *
+     * @param OutputInterface $output
+     */
+    protected function outputConfiguredHandlers(OutputInterface $output)
+    {
+        $output->writeln(
+            'Configured meta data handlers: ' . implode(', ', array_keys($this->configuredMetadataHandlers))
+        );
+        $output->writeln(
+            'Configured binary data handlers: ' . implode(', ', array_keys($this->configuredBinarydataHandlers))
+        );
+    }
+
+    /**
+     * Verify that the handler options have been set to meaningful values.
+     *
+     * @param mixed $fromHandlers
+     * @param mixed $toHandlers
+     * @param OutputInterface $output
+     * @return bool
+     */
+    protected function areHandlerOptionsValid(
+        $fromHandlers,
```

Comment:
Good point @glye ! An alternative could be to make it `validateHandlerOptions(InputInterface $input, OutputInterface $output)`. Maybe with a loop inside to avoid code duplications.